app/views/Moderation/SchoolPayments/UpdatePaymentModal/index.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { useMemo, useState, useCallback, useEffect } from 'react';
+import {
+    ObjectSchema,
+    PartialForm,
+    useForm,
+    createSubmitHandler,
+    getErrorObject,
+    requiredCondition,
+} from '@togglecorp/toggle-form';
+import {
+    Modal,
+    Button,
+    NumberInput,
+    DateInput,
+    SelectInput,
+} from '@the-deep/deep-ui';
+
+import SchoolSelectInput, { SearchSchoolType } from '#components/SchoolSelectInput';
+import NonFieldError from '#components/NonFieldError';
+
+import { Payment } from '../index';
+
+type FormType = {
+    id: string;
+    school: string;
+    date: string;
+    amount: number;
+    status: Payment['status'];
+};
+type PartialFormType = PartialForm<FormType>;
+type FormSchema = ObjectSchema<PartialFormType>;
+type FormSchemaFields = ReturnType<FormSchema['fields']>;
+
+const schema: FormSchema = {
+    fields: (): FormSchemaFields => {
+        const basicFields: FormSchemaFields = {
+            id: [],
+            school: [requiredCondition],
+            date: [requiredCondition],
+            amount: [requiredCondition],
+            status: [requiredCondition],
+        };
+        return basicFields;
+    },
+};
+
+interface StatusOption {
+    key: Payment['status'];
+    value: string;
+}
+const statusOptions: StatusOption[] = [
+    {
+        key: 'verified',
+        value: 'Verified',
+    },
+    {
+        key: 'pending',
+        value: 'Pending',
+    },
+    {
+        key: 'canceled',
+        value: 'Canceled',
+    },
+];
+
+function statusKeySelector(status: StatusOption) {
+    return status.key;
+}
+
+function statusLabelSelector(status: StatusOption) {
+    return status.value;
+}
+
+interface Props {
+    onUpdateSuccess: () => void;
+    paymentDetails: Payment | undefined | null;
+    onModalClose: () => void;
+}
+
+function UpdatePaymentModal(props: Props) {
+    const {
+        onModalClose,
+        onUpdateSuccess,
+        paymentDetails,
+    } = props;
+
+    const [schoolOptions, setSchoolOptions] = useState<SearchSchoolType[] | undefined | null>([]);
+    const initialValue: PartialFormType = useMemo(() => ({
+        id: paymentDetails?.id,
+        school: paymentDetails?.school.id,
+        date: paymentDetails?.date,
+        amount: paymentDetails?.amount,
+        status: paymentDetails?.status,
+    }), [paymentDetails]);
+
+    useEffect(() => {
+        if (paymentDetails?.school) {
+            setSchoolOptions([paymentDetails.school]);
+        }
+    }, [paymentDetails]);
```

Comment:
Any reason to add this use effect?

Reply:
otherwise, when editing the modal, the label for school is not visible.